### PR TITLE
Add icon support to eframe

### PR DESCRIPTION
**Pull request by phoglund**
Example code using this (uses the image crate to load images):

```
pub fn load_icon(icon_bytes: &Vec<u8>) -> Option<epi::IconData> {
    if let Ok(image) = image::load_from_memory(icon_bytes) {
        let image = image.to_rgba8();
        let (width, height) = image.dimensions();
        Some(epi::IconData {
            width,
            height,
            rgba: image.as_raw().to_vec(),
        })
    } else {
        None
    }
}

impl epi::App for EmbarkSubmitApp {
     ...
    fn icon_data(&self) -> Option<epi::IconData> {
        let icon_bytes = include_bytes!("../icons/my_icon.png");
        crate::icon::load_icon(&icon_bytes.to_vec())
    }
```

### CODE REVIEW

File: egui_glium/src/backend.rs [hunk 1]
```diff
@@ -131,6 +133,17 @@ fn integration_info(
...
+fn load_icon(icon_data: Option<epi::IconData>) -> Option<glutin::window::Icon> {
+    let icon_data = icon_data?;
+    if let Ok(icon) =
+        glutin::window::Icon::from_rgba(icon_data.rgba, icon_data.width, icon_data.height)
+    {
+        Some(icon)
+    } else {
+        None
+    }
+}
```
Comment (emilk) [inline egui_glium/src/backend.rs:145]:
```suggestion
    glutin::window::Icon::from_rgba(icon_data.rgba, icon_data.width, icon_data.height).ok()
```

**Review by emilk — CHANGES_REQUESTED** [review]
Very cool, thank you!

Please add a line in `eframe/CHANGELOG.md` too

**Review by emilk — APPROVED** [review]

**Comment by emilk** [comment]
Thank you!

**Comment by felipetesc** [comment]
Just another sample (tested on Windows 10):
```rust
use eframe::{egui};
use eframe::epi::IconData;
use image;

#[allow(dead_code)]
pub struct Image {
    size: (usize, usize),
    pixels: Vec<egui::Color32>,
}

#[allow(dead_code)]
impl Image {
    pub fn open_icon_data(path : &str) -> std::option::Option<IconData>{
        
        let image_buffer = image::open(path).unwrap();
        let img = image_buffer.to_rgba8();
        let size = (img.width() as u32, img.height() as u32);
        let pixels = img.into_vec();
        let icon_data = eframe::epi::IconData{
            rgba: pixels,
            width: size.0 ,
            height: size.1,
        };
        Some(icon_data) 
    }
    pub fn open_image(path : &str) -> Option<Image>{
        
        let image_buffer = image::open(path).unwrap();
        let img = image_buffer.to_rgb8();
        let size = (img.width() as usize, img.height() as usize);
        let pixels = img.into_vec();
        let pixels = pixels
            .chunks(4)
            .map(|p| egui::Color32::from_rgba_unmultiplied(p[0], p[1], p[2], p[3]))
            .collect();
        Some(Image { size, pixels })
    }

    pub fn decode(bytes: &[u8]) -> Option<Image> {
        use image::GenericImageView;
        let image = image::load_from_memory(bytes).ok()?;
        let image_buffer = image.to_rgba8();
        let size = (image.width() as usize, image.height() as usize);
        let pixels = image_buffer.into_vec();
        assert_eq!(size.0 * size.1 * 4, pixels.len());
        let pixels = pixels
            .chunks(4)
            .map(|p| egui::Color32::from_rgba_unmultiplied(p[0], p[1], p[2], p[3]))
            .collect();

        Some(Image { size, pixels })
    }
}
```
Usage:
```rust 
#![forbid(unsafe_code)]
#![cfg_attr(not(debug_assertions), deny(warnings))] // Forbid warnings in release builds
#![warn(clippy::all, rust_2018_idioms)]

mod view;
mod resources;

//mod created at src/resources/img.rs 
use resources::img::Image;
//app.rs moved from https://github.com/emilk/egui_template/src/app.rs  to https://github.com/emilk/egui_template/src/view/app.rs
use crate::view::app::App;

//#[cfg(not(target_arch = "wasm32"))]
fn main() {
    
    let icon_data = Image::open_icon_data("favicon32x32.png");
    
    let app = App::default();

    let mut native_options = eframe::NativeOptions::default();
    
    native_options.icon_data =  icon_data;
    
    eframe::run_native(Box::new(app), native_options);
}
}
```

**Comment by emilk** [comment]
@felipetesc Thanks for the example code! I think it's probably better to include the icon with `include_bytes!`, e.g. `let icon_data = Image::decode_icon_data(include_bytes!("favicon32x32.png"));`. That way you won't have any problems with working directories or forgetting to ship the icon with your binary.

**Comment by mjehrhart** [comment]
Does this work on macOS too or only windows?

**Pull request merged**